flink-end-to-end-tests/test-scripts/test_pyflink.sh

```diff
@@ -272,8 +272,6 @@ if [[ "${EXPECTED_MSG[*]}" != "${SORTED_READ_MSG[*]}" ]]; then
     exit 1
 fi
 
-stop_cluster
-
```

Comment:
I see that there is the `test_clean_up` function which also stops the cluster. How did this function call prevent the `test_clean_up` function to work properly?

Reply:
You're right. I find that the root case of this problem is that the "PyFlink YARN per-job test" in this script will cleanup the _on_exit_commands and so test_clean_up will never be called.

The calling path is as following:
```
|---start_hadoop_cluster_and_prepare_flink
        |--- source "$(dirname "$0")"/common.sh
                 |--- source "${TEST_INFRA_DIR}/common_utils.sh"
                         |--- _on_exit_commands=()
```

Will update the PR to split the "PyFlink YARN per-job test" into a separate script.

Reply:
Oh, I see. We should not source the common utils in the middle of a test!